internal/authctx/helper.go

```diff
@@ -94,6 +95,32 @@ func ProviderConfigureContext(_ context.Context, d *schema.ResourceData) (interf
 	return setContext(&config)
 }
 
+func ProviderConfigureContextWithDefaultTransport(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```

Comment:
can you explain why is this DefaultTransport needed ?

Reply:
The default transport is needed for mocking- the http mocking library I'm using can only intercept calls if they're made with the default transport.